Assets/Scripts/Layers/Adapters/DataTypeAdapters/FileTypeAdapter.cs

```diff
@@ -35,28 +36,52 @@ public void ProcessFile(string file)
         {
             if (file.EndsWith(','))
                 file = file.Remove(file.Length - 1);
+            
+            var absoluteFilePath = Path.Combine(Application.persistentDataPath, file);
+            var assetsFolderPath = Path.Combine(Application.persistentDataPath, assetsFolderName);
+            
+            if (!Directory.Exists(assetsFolderPath))
+            {
+                Directory.CreateDirectory(assetsFolderPath);
+            }
+            
+            var fileName = Path.GetFileNameWithoutExtension(file);
+            var extension = Path.GetExtension(file);
+            var newFilePathRelative = Path.Combine(assetsFolderName, fileName + extension);
+            var newFilePathAbsolute = Path.Combine(assetsFolderPath, fileName + extension);
+            
+            // Find a unique file name if a file with the same name already exists
+            int index = 0;
+            while (File.Exists(newFilePathAbsolute))
+            {
+                index++;
+                var newFileName = $"{fileName}({index}){extension}";
+                newFilePathRelative = Path.Combine(assetsFolderName, newFileName);
+                newFilePathAbsolute = Path.Combine(assetsFolderPath, newFileName);
+            }
+            
+            File.Move(absoluteFilePath, newFilePathAbsolute);
                 
-            string fileExtension = Path.GetExtension(file).ToLower();
-            if (fileExtension.StartsWith('.'))
-                fileExtension = fileExtension.Substring(1);
+            if (extension.StartsWith('.'))
+                extension = extension.Substring(1);
             
-            var fileTypeEvent = fileTypeEvents.FirstOrDefault(fte => fte.Extension == fileExtension);
+            var fileTypeEvent = fileTypeEvents.FirstOrDefault(fte => fte.Extension == extension);
```

Comment:
ik zou extentions altijd met tolower() gebrtuiken maar dat is misschien too much